docs/reference/modules/node.asciidoc

```diff
@@ -32,42 +33,52 @@ configure this setting, the node has the following roles by default:
 * `remote_cluster_client`
 * `transform`
 
-NOTE: If you set `node.roles`, the node is assigned only the roles you specify.
+[IMPORTANT]
+====
+If you set `node.roles`, ensure you specify every node role your cluster needs.
+To access most {stack} features, a minimal cluster requires nodes with the
+following roles:
+
+- `master`
+- `ingest`
+- `transform`
```

Comment:
I'm curious why `transform` but not `ml`? And what do we consider "most" Stack features? For example, is CCS here, or is that seen as advanced? If it's under the umbrella of "most" Stack features, we need `remote_cluster_client` in this list too. And now we're basically listing every role. Either way, we have almost every role listed here but not much guidance how a user decides which roles they need. I wonder if we should present it that way: use the `X` role if you're trying to do `Y` (e.g., `remote_cluster_client` if you're trying to search or replicate from another cluster, `ml` if you're trying to do analomy detection (or use `{ml-features}`), etc.)?

Reply:
Thanks for the suggestion @jasontedor. I went with this approach, which made the admonition simpler.

Rather than include a `minimal` list of roles. This now just provides guidance on the roles required for specific features.

Reply:
From reading the whole page, it seemed to me the it is written in the form "Use `X` role if you are trying to do `Y`". This is already covered.

The additional call-out box was to highlight the most obvious "getting started" roles by name. This advice could save users some time as they will want to ingest data and there is a transform that is deployed with endpoint. The user did not actively choose to load the transform and so they may not realise that they will need this role.  

It might be difficult for us to come together and agree what this initial typical set of roles would be. 

However I still think we can use our docs to anticipate problems for users who are deciding to use `node.roles`. They may be migrating from using a legacy setup where their data nodes could run transforms without additional configuration. They may have intentionally created transforms or they may just have the endpoint transform enabled.

In which case, instead of the list of typical roles, we could limit the call-out to only mention that the transform role must now be explicitly set if either the user or the Security Solution has been created one. Happy to do either, as both options help to avoid a likely configuration problem.

Reply:
Thanks @sophiec20. While the admon isn't limited to just the `transform` role, it does now highlight that you need to set the `transform` role for the Elastic Security app, which wasn't highlighted previously. This should prevent users from accidentally removing a role they didn't think was required (because they didn't know Elastic Security used transforms).

Reply:
@kevinlog Please confirm when the endpoint transform is created. It gets installed with Fleet. Are there other conditions under which it is installed?

Reply:
@sophiec20 it gets installed whenever the Endpoint Package is installed.  This happens in 2 places.

1. When navigating to Fleet as a superuser as you mentioned
2. When navigating to the Security application as a superuser

Those are the only conditions when it is installed

Reply:
Thanks @sophiec20 and @kevinlog. I've noted that Fleet also requires the `transform` node role.

Reply:
Thx LGTM